src/lib/u2.js

```diff
@@ -1437,3 +1437,52 @@ foam.CLASS({
     }
   ]
 });
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'CSS',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'code'
+    },
+    {
+      name: 'name',
+      factory: function() { return 'CSS-' + this.$UID; }
+    }
+  ],
+
+  methods: [
+    function installInClass(cls) {
+      // There might be several CSS axioms on each class.
+      // Each one will try to run this code.
+      // In order to prevent them all from running, we check if cls.create
+      // has the __cssHandled property set.
+      if ( cls.create.__cssHandled ) return;
+
+      var oldCreate = cls.create;
+      var axiom = this;
+
+      cls.create = function(args, X) {
+        // Install our own CSS, and then all parent models as well.
+        var c = this;
+        while ( c && c.private_ ) {
+          if ( X.installedCSS[c.id] ) break;
```

Comment:
I would rather see the installed check and install method combined.  I shouldn't have to know the mechanism.  Like a maybeInstallCSS().  Or, just a registerClass(), which could do more than one thing, like install CSS and global document listeners or install gesture manager.  It could just be a isFirstInstanceInDocument() method, then the implementation could be entirely in the Axiom, rather than shared between the Axiom and Window.